client/src/Components/Common/RecentArtSlider/RecentArtSlider.js

```diff
@@ -0,0 +1,86 @@
+import React , {useEffect} from 'react';
+import { useTheme } from '@material-ui/core/styles';
+import MobileStepper from '@material-ui/core/MobileStepper';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
+import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
+import SwipeableViews from 'react-swipeable-views';
+import { autoPlay } from 'react-swipeable-views-utils';
+import { Grid } from '@material-ui/core'
+import axios from 'axios'
+import useStyles from './sliderStyle'
+
+const AutoPlaySwipeableViews = autoPlay(SwipeableViews);
+
+function SwipeableTextMobileStepper() {
+  const classes = useStyles();
+  const theme = useTheme();
+  const [activeStep, setActiveStep] = React.useState(0);
+  const [recentArt , setRecentArt] = React.useState(null)
+  const [error , setError] = React.useState(null)
+  const maxSteps =3
+
+  useEffect(()=>{
+      if(recentArt){
+         return
+      }
+       else {
+        axios.get('/api/get-art')
+        .then(res => setRecentArt(res.data))
+        .catch(err => setError(err))
+       }
```

Comment:
Love the early return 💚 

One of the advantages to early returns is that it makes it easier to avoid nesting. 

Nested conditionals can get out of hand very easily, so early returns offer a great advantage! 

In this case, you can avoid nesting the code in else because the return statement functions the same way else does.



Best practices around conditionals is a subject that has a lot of opinions. Some people recommend avoiding `else`, and some even try to avoid `if` -- favoring ternary statements and hash-maps. 

I recommend doing what makes sense to you + your team -- there are usually more important/fun things to think about 😄  

---

https://medium.com/better-programming/are-early-returns-any-good-eed4b4d03866
https://medium.com/hackernoon/rethinking-javascript-the-if-statement-b158a61cd6cb

Reply:
great  :+1:  ,I was about to ask for feedback
thanks